core/src/test/java/org/apache/iceberg/jdbc/TestJdbcTableConcurrency.java

```diff
@@ -0,0 +1,148 @@
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.jdbc;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.iceberg.CatalogProperties;
+import org.apache.iceberg.DataFile;
+import org.apache.iceberg.DataFiles;
+import org.apache.iceberg.FileFormat;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.relocated.com.google.common.collect.Iterables;
+import org.apache.iceberg.relocated.com.google.common.util.concurrent.MoreExecutors;
+import org.apache.iceberg.types.Types;
+import org.apache.iceberg.util.Tasks;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import static org.apache.iceberg.TableProperties.COMMIT_MAX_RETRY_WAIT_MS;
+import static org.apache.iceberg.TableProperties.COMMIT_MIN_RETRY_WAIT_MS;
+import static org.apache.iceberg.TableProperties.COMMIT_NUM_RETRIES;
+import static org.apache.iceberg.types.Types.NestedField.required;
+
+public class TestJdbcTableConcurrency {
+
+  static final TableIdentifier TABLE_IDENTIFIER = TableIdentifier.of("db", "test_table");
+  static final Schema SCHEMA = new Schema(
+          required(1, "id", Types.IntegerType.get(), "unique ID"),
+          required(2, "data", Types.StringType.get())
+  );
+  @Rule
+  public TemporaryFolder temp = new TemporaryFolder();
+  File tableDir;
+
+  @Test
+  public synchronized void testConcurrentFastAppends() throws IOException {
+    Map<String, String> properties = new HashMap<>();
+    this.tableDir = temp.newFolder();
+    properties.put(CatalogProperties.WAREHOUSE_LOCATION, tableDir.getAbsolutePath());
+    properties.put(CatalogProperties.HIVE_URI, "jdbc:h2:mem:concurentFastAppend;create=true");
+    JdbcCatalog catalog = new JdbcCatalog();
+    catalog.setConf(new Configuration());
+    catalog.initialize("jdbc", properties);
+    catalog.createTable(TABLE_IDENTIFIER, SCHEMA);
+
+    Table icebergTable = catalog.loadTable(TABLE_IDENTIFIER);
+
+    String fileName = UUID.randomUUID().toString();
+    DataFile file = DataFiles.builder(icebergTable.spec())
+            .withPath(FileFormat.PARQUET.addExtension(fileName))
```

Comment:
Nit: continuation indents should be 2 indents, which is 4 spaces, not 8.

Reply:
updated my editor setting and reformatted files.